gin_lsat_middleware/gin_lsat_middleware.go

```diff
@@ -0,0 +1,122 @@
+package gin_lsat_middleware
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"proxy/ln"
+	"proxy/lsat"
+	macaroonutils "proxy/macaroon"
+	"proxy/service"
+	"proxy/utils"
+
+	"github.com/gin-gonic/gin"
+	"github.com/joho/godotenv"
+	"github.com/lightningnetwork/lnd/lnrpc"
+)
+
+const (
+	LND_CLIENT_TYPE   = "LND"
+	LNURL_CLIENT_TYPE = "LNURL"
+)
+
+type GinLsatMiddleware struct {
+	Response func(c *gin.Context, code int, message string)
+	Service  *service.Service
+}
+
+func NewLsatMiddleware(mw *GinLsatMiddleware) (*GinLsatMiddleware, error) {
```

Comment:
it would also be good to accept config options here instead that all the configuration for the middleware can be passed in. 

e.g. 

```

lnClient = ginlsat.InitLnClient({
  Address: "...",
  MacaroonHex: "...",
  ..
}
lsatmiddleware, err := ginlsat.NewMiddleware({
  Amount: 100,
  lnClient: lnClient,
})
```

this way it is easily configurable for the developer/user and it gives flexibility to the developer.